src/cplscheme/config/CouplingSchemeConfiguration.cpp

```diff
@@ -979,24 +979,14 @@ void CouplingSchemeConfiguration::addDataToBeExchanged(
     const bool requiresInitialization = exchange.requiresInitialization;
     if (from == accessor) {
       scheme.addDataToSend(exchange.data, exchange.mesh, requiresInitialization);
-      if (requiresInitialization && (_config.type == VALUE_SERIAL_EXPLICIT || _config.type == VALUE_SERIAL_IMPLICIT)) {
-        if (not _experimental) {
-          PRECICE_CHECK(not scheme.doesFirstStep(),
-                        "In serial coupling only second participant can initialize data and send it. "
-                        "Please check the <exchange data=\"{}\" mesh=\"{}\" from=\"{}\" to=\"{}\" initialize=\"{}\" /> tag in the <coupling-scheme:... /> of your precice-config.xml. You can bypass this restriction by activating the experimental interface for time interpolation. Please refer to the documentation for more details.",
-                        dataName, meshName, from, to, requiresInitialization);
-        }
-      }
     } else if (to == accessor) {
-      scheme.addDataToReceive(exchange.data, exchange.mesh, requiresInitialization);
-      if (requiresInitialization && (_config.type == VALUE_SERIAL_EXPLICIT || _config.type == VALUE_SERIAL_IMPLICIT)) {
-        if (not _experimental) {
-          PRECICE_CHECK(scheme.doesFirstStep(),
-                        "In serial coupling only first participant can receive initial data. "
-                        "Please check the <exchange data=\"{}\" mesh=\"{}\" from=\"{}\" to=\"{}\" initialize=\"{}\" /> tag in the <coupling-scheme:... /> of your precice-config.xml. You can bypass this restriction by activating the experimental interface for time interpolation. Please refer to the documentation for more details.",
-                        dataName, meshName, from, to, requiresInitialization);
-        }
+      // Checks for a serial coupling scheme, where initial data is received by first participant.
+      if (!scheme.doesFirstStep() && requiresInitialization && (_config.type == VALUE_SERIAL_EXPLICIT || _config.type == VALUE_SERIAL_IMPLICIT) && getWaveformUsedOrder(dataName) == 0) // the order of the statements is crucial! getWaveformUsedOrder should be called last, because it may run into an error, if the data is initialized, but not defined as read data (for example Integration/Serial/AitkenAcceleration)
```

Comment:
This part is a bit tricky: Usually the second participant to which we are writing the initial data defines this as `read-data`. This is strictly not required (@uekerman, we briefly discussed this in the chat), but I think in most of the cases everything should work.

There is the highly theoretical case that a user does not define data that is initialized as read data. If this every happens, the error in `getWaveformUsedOrder` will be raised (and this might also be something worth knowing...).